lib/form_personal_details.dart

```diff
@@ -0,0 +1,200 @@
+                hintText: 'First Name',
+                controller: _firstNameController,
+                keyboardType: TextInputType.name,
+              ),
+              // FIRST NAME ERROR
+              if (_firstNameError.isNotEmpty) ...[
+                const SizedBox(height: 16),
+                Text(
+                  _firstNameError,
+                  style: const TextStyle(color: Colors.red, fontSize: 16),
+                ),
+              ],
+              const SizedBox(height: 16),
+              FormBox(
+                icon: Icons.person_2_outlined,
+                hintText: 'Last Name',
+                controller: _lastNameController,
+                keyboardType: TextInputType.name,
+              ),
+              // LAST NAME ERROR
+              if (_lastNameError.isNotEmpty) ...[
+                const SizedBox(height: 16),
+                Text(
+                  _lastNameError,
+                  style: const TextStyle(color: Colors.red, fontSize: 16),
+                ),
+              ],
+              const SizedBox(height: 16),
+              FormBox(
+                icon: Icons.calendar_month_outlined,
+                hintText: 'Date Of Birth',
+                controller: _dateOfBirthController,
+                keyboardType: TextInputType.number,
+              ),
+              // DATE OF BIRTH ERROR
+              if (_dateOfBirthError.isNotEmpty) ...[
+                const SizedBox(height: 16),
+                Text(
+                  _dateOfBirthError,
+                  style: const TextStyle(color: Colors.red, fontSize: 16),
+                ),
+              ],
+              const SizedBox(height: 16),
+              FormBox(
+                icon: Icons.phone_outlined,
+                hintText: 'Mobile Number',
+                controller: _mobileNumberController,
+                keyboardType: TextInputType.phone,
+                // TODO: Use input formatters to modify keyboard interface?
+                // inputFormatters: <TextInputFormatter>[
+                //   FilteringTextInputFormatter.digitsOnly,
+                //   LengthLimitingTextInputFormatter(12), // Adjust according to your phone number format
+                //   also try MaskTextInputFormatter to enforce a specific format
+                // ],
+              ),
+              // MOBILE NUMBER ERROR
+              if (_mobileNumberError.isNotEmpty) ...[
+                const SizedBox(height: 16),
+                Text(
+                  _mobileNumberError,
+                  style: const TextStyle(color: Colors.red, fontSize: 16),
+                ),
+              ],
+              const SizedBox(height: 16),
+              _createNextButton,
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  // TODO: This should theoretically make the below code cleaner but the 'required' text doesn't show...
+  // void updateErrorState(TextEditingController controller, String errorField) {
```

Comment:
this is not working because you are just changing the variable errorField and not the value it holds.  You can ask me on discord if this is still confusing

Reply:
An idea would be to return a String in this function and then do something like:

firstNameError = updateErrorState(TextEditingController controller) ...